<launch>
	<arg name="package_name" value="weaving_inspection" />

	<include file="$(find rosbridge_server)/launch/rosbridge_websocket.launch" />

	<node name="weaving_inspection_main_node" pkg="$(arg package_name)" type="weaving_inspection_main" output="screen">

		<!-- Are we in simulation mode or real camera is connected-->
		<param name="simulation_mode_on" value="false" />

		<!-- Where to pick the images in simulation mode -->
		<param name="simulation_images_path" value="/home/harshit/catkin_ws/src/files_inspection/nn/set_04_clean_flat/" />
		<param name="simulation_images_path_prefix" value="" />
		<param name="simulation_images_path_suffix" value=".png" />

		<!--Inference file paths-->
		<param name="cfg_file_path" value="$(find weaving_inspection)/nn/weaving_inspection.cfg" />
		<param name="names_file_path" value="$(find weaving_inspection)/nn/weaving_inspection.names" />
		<param name="weights_file_path" value="$(find weaving_inspection)/nn/weaving_inspection_best.weights" />
		<param name="theshold_value" value="0.01" />
   
		<!-- Cropping, Resizing --> 
		<param name="resize_image_enabled" value="false" />
		<param name="resize_width" value="640" />
		<param name="resize_height" value="512" />

		<param name="crop_image_enabled" value="true" />
		<param name="crop_row" value="108" />
		<param name="crop_col" value="308" />
		<param name="crop_width" value="300" />
		<param name="crop_height" value="300" />

		<!-- Published Topics -->
		<param name="image_transport_topic_name" value="/gui/image" />
		<param name="current_result_publish_topic" value="/gui/label/inspection_category" />
		<param name="ok_count_topic_name" value="/gui/label/ok_count" />
		<param name="ng_count_topic_name" value="/gui/label/ng_count" />
		<param name="total_count_topic_name" value="/gui/label/total_count" />
		<param name="machine_status_topic_name" value="/gui/label/machine_status" />
		<param name="defect_type_count_array_topic_name" value="/gui/chart/defect_data"/> 

		<!-- Subscribed Topics -->
		<param name="start_button_topic_name" value="/gui/button/start" />
		<param name="stop_button_topic_name" value="/gui/button/stop" />
		<param name="pause_button_topic_name" value="/gui/button/pause" />
		<param name="save_image_flag_topic_name" value="/gui/checkbox/save_image" />
		<param name="threshold_value_topic_name" value="/gui/value/threshold" />

		<!-- When saving images, where do we want to save? -->
		<param name="save_image_base_path" value="/media/robro/33AB-DCCB/tetrapack_images/" />
	</node>
</launch>

 
